PSPDFKitPlugin/PSPDFKitPlugin.m

```diff
@@ -1622,4 +1621,77 @@ - (void)setFormFieldValue:(CDVInvokedUrlCommand *)command {
         }
     }
 }
+
+#pragma mark - XFDF
+
+- (void)importXFDF:(CDVInvokedUrlCommand *)command {
+    NSString *xfdfFilePath = [command argumentAtIndex:0];
+    // Validate the XFDF file path.
+    if (xfdfFilePath.length == 0) {
+        [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"The XFDF path must be a valid string."] callbackId:command.callbackId];
+        return;
+    }
+
+    NSURL *xfdfFileURL = [self fileURLWithPath:xfdfFilePath];
+    if (![NSFileManager.defaultManager fileExistsAtPath:(NSString *)xfdfFileURL.path]) {
+        [self.commandDelegate sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"The XFDF file does not exisit."] callbackId:command.callbackId];
+        return;
+    }
+
+    PSPDFDocument *document = self.pdfController.document;
+    VALIDATE_DOCUMENT(document)
+
+    PSPDFFileDataProvider *dataProvider = [[PSPDFFileDataProvider alloc] initWithFileURL:xfdfFileURL];
+    PSPDFXFDFParser *parser = [[PSPDFXFDFParser alloc] initWithDataProvider:dataProvider documentProvider:document.documentProviders[0]];
+
+    NSError *error;
+    if (![parser parseWithError:&error]) {
+        CDVPluginResult *pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR
+                                                      messageAsDictionary:@{@"localizedDescription": error.localizedDescription, @"domin": error.domain}];
+        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+        return;
+    }
+
+    // Import annotations to the document.
+    NSArray <PSPDFAnnotation *> *annotations = parser.annotations;
+    if (annotations) {
+        [document addAnnotations:annotations options:nil];
+    }
+}
+
+- (void)exportXFDF:(CDVInvokedUrlCommand *)command {
+    NSString *xfdfFilePath = [command argumentAtIndex:0];
+    // Validate the XFDF file path.
+    if (xfdfFilePath.length == 0) {
```

Comment:
Can we skip this validation step and create them in some default path like "documents/xfdf/annotations.xfdf"?

Reply:
I would force the user to give us a valid path, so I'd leave it as-is. To me this would look like too much "automagick" :)